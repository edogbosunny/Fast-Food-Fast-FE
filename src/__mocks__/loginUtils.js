/* eslint-disable import/prefer-default-export */
export const props = {
  auth: {
    isAuthenticated: true,
    loading: null,
    user: { id: 1, iat: 1549439277, exp: 1549525677 },
  },
  errors: {
    isAuthenticated: false,
    loading: null,
    email: '',
    password: '',
    message: '',
  },
  user: {},
  history: {
    action: 'PUSH',
    push: jest.fn(),
  },
  location: {
    hash: '',
    pathname: '/login',
    search: '',
  },
  loginUser: jest.fn(),
  match: { path: '/login', url: '/login', isExact: true },
};

export const mockedLoginState = {
  auth: {
    isAuthenticated: true,
    loading: null,
    user: { id: 1, iat: 1549439277, exp: 1549525677 },
  },
  errors: {
    isAuthenticated: false,
    loading: null,
    email: '',
    password: '',
    message: '',
  },
};

export const nextProps = {
  auth: {
    isAuthenticated: true,
    loading: null,
    user: { id: 1, iat: 1549439277, exp: 1549525677 },
  },
  history: {
    action: 'PUSH',
    push: jest.fn(),
  },
};

export const Loginstate = {
  email: '',
  password: '',
};
