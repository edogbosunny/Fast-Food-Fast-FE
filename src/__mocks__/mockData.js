const mock = {
  loginDetails: {
    email: 'test@test.com',
    password: '123456',
  },
  registerDetails: {
    email: 'test1@test.com',
    password: 123456,
    confirmpassword: 123456,
    firstname: 'test',
    lastname: 'test,',
  },
  errRegisterDetails: {
    email: 'test1@test.com',
    password: 123456,
    confirmpassword: 123456,
    firstname: 'test',

  },
};
export default mock;
