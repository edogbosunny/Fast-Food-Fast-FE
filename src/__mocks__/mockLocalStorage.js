let localStorage = {};

export default {
  setItem(key, value) {
    return Object.assign(localStorage, { [key]: value });
  },
  removeItem(key) {
    return delete localStorage[key];
  },
  getItem(key) {
    return localStorage[key];
  },
  clear() {
    localStorage = {};
  },
};
