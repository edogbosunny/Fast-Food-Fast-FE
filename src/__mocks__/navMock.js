/* eslint-disable import/prefer-default-export */
export const props = {
  auth: {
    isAuthenticated: true,
    loading: null,
    user: { id: 1, iat: 1549439277, exp: 1549525677 },
  },
  errors: {
    isAuthenticated: false,
    loading: null,
    email: '',
    password: '',
    message: '',
  },
  history: {
    action: 'PUSH',
    push: jest.fn(),
  },
  location: {
    hash: '',
    pathname: '/login',
    search: '',
  },
  logoutUser: jest.fn(),
  match: { path: '/login', url: '/login', isExact: true },
};

export const mockState = {
  auth: {
    isAuthenticated: true,
    loading: null,
    user: { id: 1, iat: 1549439277, exp: 1549525677 },
  },
};
