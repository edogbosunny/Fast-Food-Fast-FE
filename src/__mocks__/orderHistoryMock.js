/* eslint-disable import/prefer-default-export */
export const props = {
  dispatch: jest.fn(),
  orderHistory: {
    loading: false,
    orderHistory: {
      count: 30,
      data: [{}],
      message: 'Food order retrieved succesfully',
    },
  },
};

export const tableProps = {
  getAllOrders: jest.fn(),
  orderHistory: {
    loading: false,
    orderHistory: {
      count: 30,
      data: [{}],
      message: 'Food order retrieved succesfully',
    },
  },
};
