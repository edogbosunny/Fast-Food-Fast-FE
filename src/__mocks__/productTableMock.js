/* eslint-disable import/prefer-default-export */
export const props = {
  cart: {
    count: 800,
    cart: {
      0: {
        meal_id: 1, price: '600', meal: 'Spaniish Noodles', qty: 1,
      },
    },
    items: null,
    cartCount: 800,
  },
  checkoutOrder: jest.fn(),
  order: {
    order: {},
  },
  history: {
    push: jest.fn(),
  },
  removeFromCart: jest.fn(),
};
