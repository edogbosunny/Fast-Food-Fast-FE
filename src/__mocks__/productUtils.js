/* eslint-disable import/prefer-default-export */
export const props = {
  addToCart: jest.fn(),
  auth: {
    isAuthenticated: true,
    loading: false,
    user: { id: 1, iat: 1549456186, exp: 1549542586 },
  },
  foodMenu: {
    foodMenu: {
      count: 3,
      data: [
        { meal_id: 1, price: '600', meal: 'Spaniish Noodles' },
        { meal_id: 2, price: '200', meal: 'DODO' },
        { meal_id: 3, price: '500', meal: 'Ewagoin' },
      ],
      message: 'Food menu has been retrieved succesfully',
    },
  },
  getAllMenu: jest.fn(),
  history: {
    push: jest.fn(),
  },
  alert: {
    show: jest.fn(),
    error: jest.fn(),
    success: jest.fn(),
  },
  loader: { loading: false },
  mealQuantity: undefined,
  staticContext: undefined,
};
