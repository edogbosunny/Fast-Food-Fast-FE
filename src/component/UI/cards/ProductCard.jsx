/* eslint-disable react/destructuring-assignment */
/* eslint-disable react/prop-types */
/* eslint-disable no-shadow */
/* eslint-disable react/prefer-stateless-function */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
// import PropTypes from 'prop-types';
import { getAllMenu } from '../../../actions/foodMenuReducer';
import { addToCart } from '../../../actions/cartActions';
import '../../../styles/ind.css';
import Spinner from '../../../utils/Spinner';

class ProductCard extends Component {
  constructor() {
    super();
    this.state = {
      mealQuantity: 0,
    };
    this.onChange = this.onChange.bind(this);
    this.renderItems = this.renderItems.bind(this);
    console.log(this.state, '----state-->');
  }

  componentDidMount() {
    this.props.getAllMenu();
  }

  onAdd(menu) {
    this.props.addToCart(menu);
  }

  onChange(e) {
    this.setState({ [e.target.name]: e.target.value });
    // console.log(this.state);
  }

  renderItems() {
    const { data } = this.props.foodMenu.foodMenu;

    return data.map(Foodmeal => (
      <div key={Foodmeal.meal_id}>
        {/* cart cards would be here */}
        <div className='card griditem'>
          <div>
            <div className='cardimg cardImage' />
          </div>

          <div>
            <h5 id='meal'>
              Name:
              {' '}
              {Foodmeal.meal}
            </h5>
            <h5 id='price'>
              Price:
              {' '}
              {Foodmeal.price}
            </h5>
          </div>
          <div className='cart-btn-mag'>
            <button type='button' onClick={() => { this.onAdd(Foodmeal); }} className='btnaccept' id='id'>Add to Cart </button>
            { /* <button type='button' id='sub' className='btncomp'>-</button> */ }
            { /* <input type='number'
            onChange={this.onChange}
            id='num' className='btncomp1' name='mealQuantity' disabled /> */ }
            {/* <button type='button' onClick={() =>
             { this.onAdd(Foodmeal); }} id='add' className='btndeny'>+</button> */}
          </div>
        </div>
      </div>
    ));
  }

  render() {
    const { loader } = this.props;
    const { loading } = loader;
    console.log(this.props);
    if (loading) {
      return (<Spinner />);
    }

    return (
      <div align="center" className="gridcontainer">
        {this.renderItems()}
      </div>
    );
  }
}

const mapStateToProps = state => ({
  auth: state.auth,
  foodMenu: state.foodMenu,
  loader: state.loader,
  mealQuantity: state.mealQuantity,
});

export default withRouter(connect(mapStateToProps, { getAllMenu, addToCart })(ProductCard));
