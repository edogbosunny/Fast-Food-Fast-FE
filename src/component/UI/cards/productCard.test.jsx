import React from 'react';
import { mount } from 'enzyme';
import { ProductCard, mapStateToProps } from './ProductCard';
import { props } from '../../../__mocks__/productUtils';

const wrapper = mount(
  <ProductCard {...props} />,
);
describe('Product Card Container Test Suit', () => {
  it('should mount the product page', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render on add menu', () => {
    const spy = jest.spyOn(ProductCard.prototype, 'onAdd');
    wrapper.find('button').at(0).simulate('click');
    expect(spy).toHaveBeenCalled();
  });

  it('should return a spinner', () => {
    wrapper.find('Spinner').exists();
  });

  it('should test mapStateToProps', () => {
    const state = mapStateToProps(props);
    expect(state).toBeTruthy();
  });
});
