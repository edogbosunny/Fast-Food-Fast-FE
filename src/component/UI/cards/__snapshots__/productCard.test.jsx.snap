// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Card Container Test Suit should mount the product page 1`] = `
<ProductCard
  addToCart={[MockFunction]}
  alert={
    Object {
      "error": [MockFunction],
      "show": [MockFunction],
      "success": [MockFunction],
    }
  }
  auth={
    Object {
      "isAuthenticated": true,
      "loading": false,
      "user": Object {
        "exp": 1549542586,
        "iat": 1549456186,
        "id": 1,
      },
    }
  }
  foodMenu={
    Object {
      "foodMenu": Object {
        "count": 3,
        "data": Array [
          Object {
            "meal": "Spaniish Noodles",
            "meal_id": 1,
            "price": "600",
          },
          Object {
            "meal": "DODO",
            "meal_id": 2,
            "price": "200",
          },
          Object {
            "meal": "Ewagoin",
            "meal_id": 3,
            "price": "500",
          },
        ],
        "message": "Food menu has been retrieved succesfully",
      },
    }
  }
  getAllMenu={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": undefined,
        },
      ],
    }
  }
  history={
    Object {
      "push": [MockFunction],
    }
  }
  loader={
    Object {
      "loading": false,
    }
  }
>
  <div
    align="center"
    className="gridcontainer"
  >
    <div
      key="1"
    >
      <div
        className="card griditem"
      >
        <div>
          <div
            className="cardimg cardImage"
          />
        </div>
        <div>
          <h5
            id="meal"
          >
            Name:
             
            Spaniish Noodles
          </h5>
          <h5
            id="price"
          >
            Price:
             
            600
          </h5>
        </div>
        <div
          className="cart-btn-mag"
        >
          <button
            className="btnaccept"
            id="id"
            onClick={[Function]}
            type="button"
          >
            Add to Cart 
          </button>
        </div>
      </div>
    </div>
    <div
      key="2"
    >
      <div
        className="card griditem"
      >
        <div>
          <div
            className="cardimg cardImage"
          />
        </div>
        <div>
          <h5
            id="meal"
          >
            Name:
             
            DODO
          </h5>
          <h5
            id="price"
          >
            Price:
             
            200
          </h5>
        </div>
        <div
          className="cart-btn-mag"
        >
          <button
            className="btnaccept"
            id="id"
            onClick={[Function]}
            type="button"
          >
            Add to Cart 
          </button>
        </div>
      </div>
    </div>
    <div
      key="3"
    >
      <div
        className="card griditem"
      >
        <div>
          <div
            className="cardimg cardImage"
          />
        </div>
        <div>
          <h5
            id="meal"
          >
            Name:
             
            Ewagoin
          </h5>
          <h5
            id="price"
          >
            Price:
             
            500
          </h5>
        </div>
        <div
          className="cart-btn-mag"
        >
          <button
            className="btnaccept"
            id="id"
            onClick={[Function]}
            type="button"
          >
            Add to Cart 
          </button>
        </div>
      </div>
    </div>
  </div>
</ProductCard>
`;
